integration/mediation-tests/tests-sample/src/test/java/org/wso2/carbon/esb/samples/test/transport/Sample254TestCase.java

```diff
@@ -61,6 +61,16 @@ public void setEnvironment() throws Exception {
         originalFolder = new File(pathToVfsDir + "original" + File.separator);
 
         FileUtils.deleteDirectory(outFolder);
+        while (outFolder.exists()) {
+            try {
+                //wait until the directory is deleted
+                TimeUnit.MILLISECONDS.sleep(100);
+                log.info("Waiting for out folder to be deleted...");
+            } catch (InterruptedException e) {
+                //Continue looping until directory is deleted
+            }
+        }
+
         FileUtils.deleteDirectory(inFolder);
         FileUtils.deleteDirectory(originalFolder);
```

Comment:
We need to wait after all 3 deletes, don't we?

Reply:
Fixed with 56d2120